pkg/apis/serving/v1alpha1/revision_lifecycle.go

```diff
@@ -80,6 +81,18 @@ func (rs *RevisionSpec) GetContainer() *corev1.Container {
 	return &corev1.Container{}
 }
 
+// GetContainerConcurrency returns the container concurrency. If
+// container concurrency is not set, the default value will be returned.
```

Comment:
I'd like to call out upgrade specifically:
```suggestion
// container concurrency is not set, the default value will be returned.
// We use the original default (0) here for backwards compatibility.
// Previous versions of Knative equated unspecified and zero, so to avoid
// changing the value used by Revisions with unspecified values when a different
// default is configured, we use the original default instead of the configured
// default to remain safe across upgrades.
```